crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -384,33 +387,45 @@ impl Format<PyFormatContext<'_>> for MaybeParenthesizeExpression<'_> {
         match needs_parentheses {
             OptionalParentheses::Multiline => match parenthesize {
                 Parenthesize::IfBreaksOrIfRequired => {
+                    let indent = !(f.context().options().preview().is_enabled()
+                        && is_expression_huggable(expression));
                     parenthesize_if_expands(&expression.format().with_options(Parentheses::Never))
+                        .with_indent(indent)
                         .fmt(f)
                 }
+
                 Parenthesize::IfRequired => {
                     expression.format().with_options(Parentheses::Never).fmt(f)
                 }
+
                 Parenthesize::Optional | Parenthesize::IfBreaks => {
                     if can_omit_optional_parentheses(expression, f.context()) {
                         optional_parentheses(&expression.format().with_options(Parentheses::Never))
                             .fmt(f)
                     } else {
+                        let indent = !(f.context().options().preview().is_enabled()
+                            && is_expression_huggable(expression));
                         parenthesize_if_expands(
                             &expression.format().with_options(Parentheses::Never),
                         )
+                        .with_indent(indent)
                         .fmt(f)
                     }
                 }
             },
             OptionalParentheses::BestFit => match parenthesize {
                 Parenthesize::IfBreaksOrIfRequired => {
+                    let indent = !(f.context().options().preview().is_enabled()
+                        && is_expression_huggable(expression));
                     parenthesize_if_expands(&expression.format().with_options(Parentheses::Never))
+                        .with_indent(indent)
                         .fmt(f)
```

Comment:
Nit: this becomes somewhat repetitive. Is it worth introducing a `parenthesize_expression_if_expands` helper?